Fail
file path is :
projects/deltachat-core/rust/src/chat.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error: incorrect use of `await`
    --> src/chat.rs:3913:9
     |
3913 |     if !await!(is_self_in_chat(context, chat_id)).await? {
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: `await` is a postfix operation: `is_self_in_chat(context, chat_id).await`

error: incorrect use of `await`
    --> src/chat.rs:3925:12
     |
3925 |         if await!(DO_SEND_STATUS_MAILS).await? {
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: `await` is a postfix operation: `DO_SEND_STATUS_MAILS.await`

error: incorrect use of `await`
    --> src/chat.rs:3928:17
     |
3928 |                 await!(set_group_explicitly_left(context, &chat.grpid)).await?;
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: `await` is a postfix operation: `set_group_explicitly_left(context, &chat.grpid).await`

error: incorrect use of `await`
    --> src/chat.rs:3942:22
     |
3942 |             msg.id = await!(send_msg(context, chat_id, msg)).await?;
     |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: `await` is a postfix operation: `send_msg(context, chat_id, msg).await`

error: incorrect use of `await`
    --> src/chat.rs:3948:9
     |
3948 |     if !await!(dc_sqlite3_execute(context.sql.clone(), q3)).await? {
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: `await` is a postfix operation: `dc_sqlite3_execute(context.sql.clone(), q3).await`

error: expected identifier, found `:`
    --> src/chat.rs:3956:8
     |
3956 | cleanup:
     |        ^ expected identifier

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3904:9
     |
3904 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3910:9
     |
3910 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3921:9
     |
3921 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3949:9
     |
3949 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error[E0425]: cannot find value `DC_CONTEXT_MAGIC` in this scope
    --> src/chat.rs:3900:29
     |
3900 |         || context.magic != DC_CONTEXT_MAGIC
     |                             ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_LAST_SPECIAL` in this scope
    --> src/chat.rs:3902:27
     |
3902 |         || (contact_id <= DC_CONTACT_ID_LAST_SPECIAL && contact_id != DC_CONTACT_ID_SELF)
     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `DC_CHAT_ID_LAST_SPECIAL`
     |
    ::: src/constants.rs:123:1
     |
123  | pub const DC_CHAT_ID_LAST_SPECIAL: ChatId = ChatId::new(9);
     | ----------------------------------------------------------- similarly named constant `DC_CHAT_ID_LAST_SPECIAL` defined here

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3902:71
     |
3902 |         || (contact_id <= DC_CONTACT_ID_LAST_SPECIAL && contact_id != DC_CONTACT_ID_SELF)
     |                                                                       ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_ERROR_SELF_NOT_IN_GROUP` in this scope
    --> src/chat.rs:3916:13
     |
3916 |             DC_EVENT_ERROR_SELF_NOT_IN_GROUP,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_MSG_TEXT` in this scope
    --> src/chat.rs:3926:23
     |
3926 |             msg.typ = DC_MSG_TEXT;
     |                       ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3927:30
     |
3927 |             if contact.id == DC_CONTACT_ID_SELF {
     |                              ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_STR_MSGGROUPLEFT` in this scope
    --> src/chat.rs:3929:57
     |
3929 |                 msg.text = dc_stock_system_msg(context, DC_STR_MSGGROUPLEFT, None, None, DC_CONTACT_ID_SELF).await?;
     |                                                         ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3929:90
     |
3929 |                 msg.text = dc_stock_system_msg(context, DC_STR_MSGGROUPLEFT, None, None, DC_CONTACT_ID_SELF).await?;
     |                                                                                          ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_STR_MSGDELMEMBER` in this scope
    --> src/chat.rs:3933:21
     |
3933 |                     DC_STR_MSGDELMEMBER,
     |                     ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3936:21
     |
3936 |                     DC_CONTACT_ID_SELF,
     |                     ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CMD_MEMBER_REMOVED_FROM_GROUP` in this scope
    --> src/chat.rs:3940:25
     |
3940 |             msg.set_cmd(DC_CMD_MEMBER_REMOVED_FROM_GROUP);
     |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_PARAM_CMD_ARG` in this scope
    --> src/chat.rs:3941:21
     |
3941 |             msg.set(DC_PARAM_CMD_ARG, contact.addr.clone());
     |                     ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_MSGS_CHANGED` in this scope
    --> src/chat.rs:3943:33
     |
3943 |             context.cb(context, DC_EVENT_MSGS_CHANGED, chat_id, msg.id).await;
     |                                 ^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_CHAT_MODIFIED` in this scope
    --> src/chat.rs:3952:25
     |
3952 |     context.cb(context, DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
     |                         ^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_get_contact` in this scope
    --> src/chat.rs:3894:19
     |
3894 |     let contact = dc_get_contact(context, contact_id).await?;
     |                   ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_chat_new` in this scope
    --> src/chat.rs:3895:16
     |
3895 |     let chat = dc_chat_new(context).await?;
     |                ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_msg_new_untyped` in this scope
    --> src/chat.rs:3896:15
     |
3896 |     let msg = dc_msg_new_untyped(context).await?;
     |               ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0599]: no method named `is_null_or_invalid` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3899:16
     |
3899 |     if context.is_null_or_invalid()
     |                ^^^^^^^^^^^^^^^^^^ method not found in `&Context`

error[E0609]: no field `magic` on type `&context::Context`
    --> src/chat.rs:3900:20
     |
3900 |         || context.magic != DC_CONTEXT_MAGIC
     |                    ^^^^^ unknown field
     |
     = note: available field is: `inner`
     = note: available fields are: `blobdir`, `sql`, `smeared_timestamp`, `generating_key_mutex`, `oauth2_mutex` ... and 17 others

error[E0425]: cannot find function `real_group_exists` in this scope
    --> src/chat.rs:3907:13
     |
3907 |     if 0 == real_group_exists(context, chat_id).await?
     |             ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_chat_load_from_db` in this scope
    --> src/chat.rs:3908:17
     |
3908 |         || 0 == dc_chat_load_from_db(chat, chat_id).await?
     |                 ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_log_event` in this scope
    --> src/chat.rs:3914:9
     |
3914 |         dc_log_event(
     |         ^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_stock_system_msg` in this scope
    --> src/chat.rs:3929:28
     |
3929 |                 msg.text = dc_stock_system_msg(context, DC_STR_MSGGROUPLEFT, None, None, DC_CONTACT_ID_SELF).await?;
     |                            ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_stock_system_msg` in this scope
    --> src/chat.rs:3931:28
     |
3931 |                 msg.text = dc_stock_system_msg(
     |                            ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0599]: no method named `cb` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3943:21
     |
3943 |             context.cb(context, DC_EVENT_MSGS_CHANGED, chat_id, msg.id).await;
     |                     ^^ method not found in `&Context`

error[E0425]: cannot find function `sqlite3_mprintf` in this scope
    --> src/chat.rs:3947:14
     |
3947 |     let q3 = sqlite3_mprintf("DELETE FROM chats_contacts WHERE chat_id=%i AND contact_id=%i;", chat_id, contact_id)?;
     |              ^^^^^^^^^^^^^^^ not found in this scope

error[E0599]: no method named `cb` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3952:13
     |
3952 |     context.cb(context, DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
     |             ^^ method not found in `&Context`

Some errors have detailed explanations: E0425, E0599, E0609.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `deltachat` (lib) due to 37 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `deltachat` (lib test) due to 37 previous errors
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

